pkg/services/ngalert/notifier/channels/dingding.go

```diff
@@ -17,139 +17,148 @@ import (
 
 const defaultDingdingMsgType = "link"
 
-type DingDingConfig struct {
-	*NotificationChannelConfig
-	MsgType string
-	Message string
-	URL     string
+type dingDingSettings struct {
+	URL         string `json:"url,omitempty" yaml:"url,omitempty"`
+	MessageType string `json:"msgType,omitempty" yaml:"msgType,omitempty"`
+	Title       string `json:"title,omitempty" yaml:"title,omitempty"`
+	Message     string `json:"message,omitempty" yaml:"message,omitempty"`
 }
 
-func NewDingDingConfig(config *NotificationChannelConfig) (*DingDingConfig, error) {
-	url := config.Settings.Get("url").MustString()
-	if url == "" {
-		return nil, errors.New("could not find url property in settings")
+func buildDingDingSettings(fc FactoryConfig) (dingDingSettings, error) {
+	settings := dingDingSettings{}
```

Comment:
I'd use this syntax for initializing a zero-value struct. Up to you.
```suggestion
	var settings dingDingSettings
```